--- a/data/org.gnome.Chess.appdata.xml.in	2019-09-13 11:53:29.248950139 +0200
+++ b/data/org.gnome.Chess.appdata.xml.in	2019-09-13 11:54:15.001599409 +0200
@@ -38,6 +38,9 @@
     <kudo>ModernToolkit</kudo>
     <kudo>UserDocs</kudo>
   </kudos>
+  <releases>
+    <release version="3.34.0" date="2019-09-08" />
+  </releases>
   <translation type="gettext">gnome-chess</translation>
   <developer_name>The GNOME Project</developer_name>
   <content_rating type="oars-1.0">
